### PR TITLE
Implementing control flow collections

### PR DESCRIPTION
**Pull request by nate-thegrate**
This pull request aims for improved readability, based on issue #146600.

```dart
// before
Set<Color> _distinctVisibleColors() {
  final Set<Color> distinctVisibleColors = <Color>{};
  if (top.style != BorderStyle.none) {
    distinctVisibleColors.add(top.color);
  }
  if (right.style != BorderStyle.none) {
    distinctVisibleColors.add(right.color);
  }
  if (bottom.style != BorderStyle.none) {
    distinctVisibleColors.add(bottom.color);
  }
  if (left.style != BorderStyle.none) {
    distinctVisibleColors.add(left.color);
  }
  return distinctVisibleColors;
}


// after
Set<Color> _distinctVisibleColors() {
  return <Color>{
    if (top.style != BorderStyle.none) top.color,
    if (right.style != BorderStyle.none) right.color,
    if (bottom.style != BorderStyle.none) bottom.color,
    if (left.style != BorderStyle.none) left.color,
  };
}
```

Most of the repo should be covered in this PR (aside from `flutter_tools/`, since there was a lot going on in there).

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

**Review by goderbauer — APPROVED** [review]
LGTM

Thank you!

**Review by Piinks — APPROVED** [review]
Flutter_LGTM

File: dev/bots/test.dart [hunk 1]
```diff
@@ -1339,28 +1335,21 @@ Future<void> verifyExist(
...
-  }
-
+  final List<String> binaryPaths = await findBinaryPaths(
+    path.join(flutterRoot, 'bin', 'cache'),
+    processManager: processManager,
+  );
   final List<String> allExpectedFiles = binariesWithEntitlements(flutterRoot) + binariesWithoutEntitlements(flutterRoot);
+  final Set<String> foundFiles = <String>{
+    for (final String binaryPath in binaryPaths)
+      if (allExpectedFiles.contains(binaryPath)) binaryPath
```
Comment (justinmc) [inline dev/bots/test.dart]: I had to double check the styleguide but looks like this is right. It should be on one line like this (as long as it fits in the line length). https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#prefer-single-line-for-short-collection-if-and-collection-for

File: dev/devicelab/lib/tasks/microbenchmarks.dart [hunk 2]
```diff
@@ -39,8 +39,8 @@ TaskFunction createMicrobenchmarkTask({
             if (enableImpeller != null && !enableImpeller) '--no-enable-impeller',
             '-d',
             device.deviceId,
+            benchmarkPath,
           ];
-          options.add(benchmarkPath);
```
Comment (justinmc) [inline dev/devicelab/lib/tasks/microbenchmarks.dart:43]: That was a weird one...

File: dev/integration_tests/android_semantics_testing/lib/src/common.dart [hunk 3]
```diff
@@ -154,14 +154,11 @@ class AndroidSemanticsNode {
...
-    for (final int id in actions) {
-      final AndroidSemanticsAction? action = AndroidSemanticsAction.deserialize(id);
-      if (action != null) {
-        convertedActions.add(action);
-      }
-    }
-    return convertedActions;
+    return <AndroidSemanticsAction>[
+      for (final int id in actions)
+        if (AndroidSemanticsAction.deserialize(id) case final AndroidSemanticsAction action)
```
Comment (justinmc) [inline dev/integration_tests/android_semantics_testing/lib/src/common.dart:159]: I didn't know about this `case` syntax.

**Review by justinmc — APPROVED** [review]
LGTM 👍  Thanks for making the codebase more readable as always.

File: packages/flutter/lib/src/foundation/timeline.dart [hunk 4]
```diff
@@ -349,16 +347,10 @@ final class _StringListChain {
...
-      }
-    }
-    for (int i = 0; i < _pointer; i++) {
-      result.add(_slice[i]!);
-    }
-    return result;
+    return <String?>[
+      for (final List<String?> slice in _chain) ...slice,
+      ..._slice.sublist(0, _pointer),
+    ].cast<String>();
```
[the remark this answers is not in the document]
Comment (nate-thegrate) [reply packages/flutter/lib/src/foundation/timeline.dart:353]: I had no idea that `.cast` worked that way, thank you!

**Comment by nate-thegrate** [comment]
Thank you all very much!

It looks like a merge conflict popped up and `autosubmit` was removed, so if a kind soul is able to re-add it, that would be much appreciated.

**Pull request merged**